src/Analyzers/CSharp/Analyzers/UsePatternCombinators/AnalyzedPattern.cs

```diff
@@ -0,0 +1,135 @@
+        /// </summary>
+        internal sealed class Source : AnalyzedPattern
+        {
+            public readonly PatternSyntax PatternSyntax;
+
+            public Source(PatternSyntax patternSyntax, IOperation target) : base(target)
+                => PatternSyntax = patternSyntax;
+        }
+
+        /// <summary>
+        /// Represents a constant-pattern, constructed from an equality check
+        /// </summary>
+        internal sealed class Constant : AnalyzedPattern
+        {
+            public readonly ExpressionSyntax ExpressionSyntax;
+
+            public Constant(ExpressionSyntax expression, IOperation target) : base(target)
+                => ExpressionSyntax = expression;
+        }
+
+        /// <summary>
+        /// Represents a relational-pattern, constructed from relational operators
+        /// </summary>
+        internal sealed class Relational : AnalyzedPattern
+        {
+            public readonly BinaryOperatorKind OperatorKind;
+            public readonly ExpressionSyntax Value;
+
+            public Relational(BinaryOperatorKind operatorKind, ExpressionSyntax value, IOperation target) : base(target)
+            {
+                OperatorKind = operatorKind;
+                Value = value;
+            }
+        }
+
+        /// <summary>
+        /// Represents an and/or pattern, constructed from a logical and/or expression.
+        /// </summary>
+        internal sealed class Binary : AnalyzedPattern
+        {
+            public readonly AnalyzedPattern Left;
+            public readonly AnalyzedPattern Right;
+            public readonly bool IsDisjunctive;
+            public readonly SyntaxToken Token;
+
+            internal Binary(AnalyzedPattern leftPattern, AnalyzedPattern rightPattern, bool isDisjunctive, SyntaxToken token, IOperation target) : base(target)
+            {
+                Left = leftPattern;
+                Right = rightPattern;
+                IsDisjunctive = isDisjunctive;
+                Token = token;
+            }
+
+            public static AnalyzedPattern? Create(AnalyzedPattern leftPattern, AnalyzedPattern rightPattern, bool isDisjunctive, SyntaxToken token)
+            {
+                var target = leftPattern.Target;
+                if (!SyntaxFactory.AreEquivalent(target.Syntax, rightPattern.Target.Syntax))
+                    return null;
+
+                return new Binary(leftPattern, rightPattern, isDisjunctive, token, target);
+            }
+        }
+
+        /// <summary>
+        /// Represents a not-pattern, constructed from inequality check or a logical-not expression.
+        /// </summary>
+        internal sealed class Not : AnalyzedPattern
+        {
+            public readonly AnalyzedPattern Pattern;
+
+            private Not(AnalyzedPattern pattern, IOperation target) : base(target)
+                => Pattern = pattern;
+
+            private static BinaryOperatorKind Negate(BinaryOperatorKind kind)
```

Comment:
do we need negating anymore?  i thought we didnt' have demorgans?

Reply:
we won't "synthesize" `not` patterns to apply demorgan, but if we have `!(x > 0)`, `!(x != 0)` or the like in source, we'll simplify.

Reply:
got it.  thanks :)